kotest-assertions/src/commonMain/kotlin/io/kotest/matchers/iterator/IteratorMatchers.kt

```diff
@@ -0,0 +1,19 @@
+package io.kotest.matchers.iterator
+
+import io.kotest.Matcher
+import io.kotest.MatcherResult
+import io.kotest.assertions.stringRepr
+import io.kotest.should
+import io.kotest.shouldNot
+
+fun <T> Iterator<T>.shouldBeEmpty() = this should beEmpty()
+fun <T> Iterator<T>.shouldNotBeEmpty() = this shouldNot beEmpty()
+fun <T> beEmpty(): Matcher<Iterator<T>> = object : Matcher<Iterator<T>> {
+   override fun test(value: Iterator<T>): MatcherResult {
+      return MatcherResult(
+         !value.hasNext(),
+         "Iterable should be empty but next value is ${stringRepr(value.next())}",
```

Comment:
@sksamuel There is a potential here for an error to be thrown if `hasNext()` is true but `value.next()` has no value and throws an `NoSuchElementException`. It feels like that's a misconfigured iterator, but since its feasible, any thoughts on what to do?

Reply:
I think that by `Iterator` specification, it's impossible by contract to have a `hasNext` return true but no `next`. I think we ignore this kind of scenario